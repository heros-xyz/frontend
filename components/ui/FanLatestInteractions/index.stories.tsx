import type { Meta, StoryObj } from "@storybook/react";
import FanLatestInteractions from ".";

const meta: Meta<typeof FanLatestInteractions> = {
  title: "Fan/FanLatestInteractions",
  component: FanLatestInteractions,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
  argTypes: {},
};

export default meta;
type Story = StoryObj<typeof FanLatestInteractions>;

export const FanInteractions: Story = {
  args: {
    titleHeading: "Latest interactions",
    actionText: "view all",
    items: [
      {
        id: "1",
        content: "hehe",
        user: {
          id: "1",
          avatar: "/images/small_img_athletes.png",
        },
        interactionMedia: [
          {
            id: "12345",
            url: "/images/small_img_athletes.png",
            createdAt: "",
            extension: "",
            sortOrder: 2,
            type: "",
            updatedAt: "",
          },
          {
            id: "12345",
            url: "/images/small_img_athletes.png",
            createdAt: "",
            extension: "",
            sortOrder: 3,
            type: "",
            updatedAt: "",
          },
        ],
      },
      {
        id: "1",
        content: "hehe",
        user: {
          id: "1",
          avatar: "/images/small_img_athletes.png",
        },
        interactionMedia: [
          {
            id: "12345",
            url: "/images/small_img_athletes.png",
            createdAt: "",
            extension: "",
            sortOrder: 3,
            type: "",
            updatedAt: "",
          },
        ],
      },
    ],
  },
};
