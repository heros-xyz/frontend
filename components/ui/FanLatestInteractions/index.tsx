import {
  Box,
  Heading,
  Link,
  Text,
  Image,
  AspectRatio,
  Grid,
  GridItem,
  Skeleton,
} from "@chakra-ui/react";
import React from "react";
import NextLink from "next/link";
import { Else, If, Then } from "react-if";
import { IconArrowRight } from "@/components/svg/IconArrowRight";
import { IconMessage } from "@/components/svg/IconMessage";
import { ILatestInteraction } from "@/types/athlete/types";
import { getImageLink } from "@/utils/link";
import { PlayVideoIcon } from "@/components/svg/PlayVideoIcon";

interface FanInteractionsProps {
  titleHeading: string;
  actionText: string;
  items: ILatestInteraction[];
  isLoading: boolean;
}

const FanLatestInteractions: React.FC<FanInteractionsProps> = ({
  titleHeading,
  items,
  actionText = "view all",
  isLoading,
}) => {
  return (
    <Box bg="white" py="5">
      <Box
        display="flex"
        justifyContent="space-between"
        alignItems="center"
        mb={{ base: 5, lg: 7 }}
      >
        <Heading
          color="primary"
          fontWeight="bold"
          fontSize="xl"
          lineHeight="110%"
        >
          {titleHeading}
        </Heading>
        {items?.length ? (
          <Box borderBottom="1px" borderColor="grey.3">
            <Link
              as={NextLink}
              fontSize={{ base: "xs", lg: "md" }}
              fontWeight="medium"
              lineHeight="100%"
              color="grey.3"
              href="fan/interactions"
              textTransform="capitalize"
              mr="2"
              _hover={{ textDecoration: "none" }}
            >
              {actionText}
            </Link>
            <Link as={NextLink} href="fan/interactions">
              <IconArrowRight width="3" height="11" color="grey.3" />
            </Link>
          </Box>
        ) : (
          ""
        )}
      </Box>
      <If condition={!isLoading}>
        <Then>
          {" "}
          {items?.length ? (
            <Box
              display="grid"
              gridTemplateColumns="repeat(3, 1fr)"
              gap="4"
              pb={4}
            >
              {items?.map((item, index) => (
                <Link
                  as={NextLink}
                  position="relative"
                  key={`${"item" + index}`}
                  cursor="pointer"
                  href={`/fan/athlete-profile/${item?.user?.id}/interaction?view=${item?.id}`}
                >
                  <AspectRatio ratio={10 / 12} w="full">
                    <Box
                      borderRadius="10"
                      bg="transparent"
                      overflow="hidden"
                      objectFit="cover"
                      borderWidth={{
                        base: `${
                          !item.interactionMedia?.[0]?.url ? "1px" : ""
                        }`,
                        lg: `${!item.interactionMedia?.[0]?.url ? "2px" : ""}`,
                      }}
                      borderColor={`${
                        !item.interactionMedia?.[0]?.url ? "secondary" : ""
                      }`}
                      w="full"
                      h="full"
                      position="relative"
                    >
                      {item.interactionMedia?.[0]?.url ? (
                        <If
                          condition={
                            item.interactionMedia?.[0]?.type === "image"
                          }
                        >
                          <Then>
                            <Image
                              src={getImageLink(
                                item.interactionMedia?.[0]?.url
                              )}
                              alt="heros item"
                              fallbackSrc="https://via.placeholder.com/150x200"
                              w="full"
                              h="full"
                              objectFit="cover"
                            />
                          </Then>
                          <Else>
                            <Box
                              position="relative"
                              w="full"
                              h="full"
                              rounded="8px"
                            >
                              <video
                                src={getImageLink(
                                  item?.interactionMedia[0]?.url
                                )}
                                style={{
                                  borderRadius: "8px",
                                  width: "100%",
                                }}
                              />
                              <PlayVideoIcon
                                w={{ base: "30px", lg: "35px" }}
                                h={{ base: "30px", lg: "35px" }}
                                position="absolute"
                                top="50%"
                                left="50%"
                                transform="translate(-50%,-50%)"
                              />
                            </Box>
                          </Else>
                        </If>
                      ) : (
                        <>
                          <Box
                            position="absolute"
                            left="14.18px"
                            top={{ base: 4, lg: 6 }}
                            h="5"
                          >
                            <IconMessage
                              width={{ base: "22px", lg: "24px" }}
                              height="5"
                              color="secondary"
                            />
                          </Box>
                          <Text
                            position="absolute"
                            left="-25px"
                            top={{ base: "45px", lg: "55px" }}
                            bottom="auto"
                            transform="translate(50%, 0)"
                            color="secondary"
                            fontSize={{ base: "xxs", lg: "md" }}
                            fontWeight="medium"
                            lineHeight="120%"
                            width="20"
                            margin="auto"
                            textOverflow="ellipsis"
                            overflow="hidden"
                            wordBreak="break-word"
                          >
                            {item.content.length > 30
                              ? `${item.content.substring(0, 30)}...`
                              : item.content}
                          </Text>
                        </>
                      )}
                    </Box>
                  </AspectRatio>

                  <Box
                    w={{ base: "10", lg: "60px" }}
                    h={{ base: "10", lg: "60px" }}
                    borderRadius="full"
                    overflow="hidden"
                    objectFit="contain"
                    border="3px"
                    borderColor="secondary"
                    borderStyle="solid"
                    position="absolute"
                    left="auto"
                    right="50%"
                    bottom={{ base: "-16px", lg: "-28px" }}
                    transform="translate(50%, 0)"
                  >
                    <Image
                      alt="avatar"
                      src={getImageLink(item.user?.avatar)}
                      fallbackSrc="https://via.placeholder.com/150"
                      w="full"
                      h="full"
                      objectFit="cover"
                    />
                  </Box>
                </Link>
              ))}
            </Box>
          ) : (
            <Box>
              <Text
                fontSize={{ base: "sm", xl: "md" }}
                color="primary"
                fontWeight="normal"
              >
                Subscribe to athletes, and you&apos;ll see their latest
                interactions here.
              </Text>
            </Box>
          )}
        </Then>
        <Else>
          <Grid
            templateColumns="repeat(3, 1fr)"
            columnGap={4}
            rowGap={{ base: 4, lg: 8 }}
          >
            <GridItem>
              <AspectRatio ratio={3 / 4}>
                <Skeleton rounded="md" />
              </AspectRatio>
            </GridItem>
            <GridItem>
              <AspectRatio ratio={3 / 4}>
                <Skeleton rounded="md" />
              </AspectRatio>
            </GridItem>
            <GridItem>
              <AspectRatio ratio={3 / 4}>
                <Skeleton rounded="md" />
              </AspectRatio>
            </GridItem>
          </Grid>
        </Else>
      </If>
    </Box>
  );
};

export default FanLatestInteractions;
