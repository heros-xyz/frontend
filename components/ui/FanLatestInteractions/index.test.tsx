import { render } from "@testing-library/react";
import FanLatestInteractions from ".";

it("renders FanLatestInteractions unchanged", () => {
  const { container } = render(
    <FanLatestInteractions
      items={[
        {
          id: "1",
          content: "123",
          user: {
            id: "1234",
            avatar: "123124",
          },
          interactionMedia: [],
        },
        {
          id: "1",
          content: "123",
          user: {
            id: "1234",
            avatar: "123124",
          },
          interactionMedia: [],
        },
      ]}
      titleHeading="Latest interactions"
      actionText="view all"
    />
  );
  expect(container).toMatchSnapshot();
});
