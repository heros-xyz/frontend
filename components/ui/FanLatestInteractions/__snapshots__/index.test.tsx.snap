// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FanLatestInteractions unchanged 1`] = `
<div>
  <div
    class="css-1pwwxxz"
  >
    <div
      class="css-n6hccp"
    >
      <h2
        class="chakra-heading css-qr4qy3"
      >
        Latest interactions
      </h2>
      <div
        class="css-14s0qel"
      >
        <a
          class="chakra-link css-1th3ek1"
          href="fan/interactions"
        >
          view all
        </a>
        <a
          class="chakra-link css-0"
          href="fan/interactions"
        >
          <svg
            class="chakra-icon css-xnm48g"
            focusable="false"
            viewBox="0 0 12 11"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 5.5L11 5.5M11 5.5L6.5 1M11 5.5L6.5 10"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
        </a>
      </div>
    </div>
     
    <div
      class="css-zxuka3"
    >
      <a
        class="chakra-link css-dfpqc0"
        href="/fan/athlete-profile/1234/interaction?view=1"
      >
        <div
          class="chakra-aspect-ratio css-1vf0n2n"
        >
          <div
            class="css-159vqpg"
          >
            <div
              class="css-15l3wnn"
            >
              <svg
                class="chakra-icon css-s04ltm"
                focusable="false"
                viewBox="0 0 22 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16.4544 5H18.6362C19.8412 5 20.818 5.89543 20.818 7V13C20.818 14.1046 19.8412 15 18.6362 15H16.4544V19L12.0907 15H7.72709C7.1246 15 6.57915 14.7761 6.18432 14.4142M6.18432 14.4142L9.90891 11H14.2725C15.4775 11 16.4544 10.1046 16.4544 9V3C16.4544 1.89543 15.4775 1 14.2725 1H3.36346C2.15847 1 1.18164 1.89543 1.18164 3V9C1.18164 10.1046 2.15847 11 3.36346 11H5.54528V15L6.18432 14.4142Z"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
              </svg>
            </div>
            <p
              class="chakra-text css-i2rf96"
            >
              123
            </p>
          </div>
        </div>
        <div
          class="css-9dkrto"
        >
          <img
            alt="avatar"
            class="chakra-image css-1gthvdk"
            src="undefined/123124"
          />
        </div>
      </a>
      <a
        class="chakra-link css-dfpqc0"
        href="/fan/athlete-profile/1234/interaction?view=1"
      >
        <div
          class="chakra-aspect-ratio css-1vf0n2n"
        >
          <div
            class="css-159vqpg"
          >
            <div
              class="css-15l3wnn"
            >
              <svg
                class="chakra-icon css-s04ltm"
                focusable="false"
                viewBox="0 0 22 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16.4544 5H18.6362C19.8412 5 20.818 5.89543 20.818 7V13C20.818 14.1046 19.8412 15 18.6362 15H16.4544V19L12.0907 15H7.72709C7.1246 15 6.57915 14.7761 6.18432 14.4142M6.18432 14.4142L9.90891 11H14.2725C15.4775 11 16.4544 10.1046 16.4544 9V3C16.4544 1.89543 15.4775 1 14.2725 1H3.36346C2.15847 1 1.18164 1.89543 1.18164 3V9C1.18164 10.1046 2.15847 11 3.36346 11H5.54528V15L6.18432 14.4142Z"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
              </svg>
            </div>
            <p
              class="chakra-text css-i2rf96"
            >
              123
            </p>
          </div>
        </div>
        <div
          class="css-9dkrto"
        >
          <img
            alt="avatar"
            class="chakra-image css-1gthvdk"
            src="undefined/123124"
          />
        </div>
      </a>
    </div>
  </div>
</div>
`;
